kratos/python_scripts/restart_utility.py

```diff
@@ -213,3 +274,29 @@ def __GetPrettyTime(self, time):
         pretty_time = "{0:.12g}".format(time)
         pretty_time = float(pretty_time)
         return pretty_time
+
+    def __IsRestartFile(self, file_name):
+        """Check whether the input file name is appropriate for a restart file."""
+        if file_name.endswith('.rest') and self.raw_file_name in os.path.basename(file_name):
+            # additional checks might have to be performed here if multiple simulations
+            # save their restart files in the same directory.
+            return True
+        return False
+
+    def __ExtractFileLabel(self, file_name):
+        """
+        Return a list of labels attached to a file name (list entries separated by '_').
+        Expecting one of two possible file name formats:
+            1) serial execution     : <file_base_name>_<step_id>.rest
+            2) parallel execution   : <file_base_name>_<thread_id>_<step_id>.rest
+        The returned list always has 2 entries, but the first component will be '' for
+        serial execution.
+        """
+        label_begin = file_name.find(self.raw_file_name + '_') + len(self.raw_file_name + '_')
+        label_end   = file_name.find('.rest')
+
+        labels      = file_name[label_begin:label_end].split("_")
+        if len(labels) < 2:
+            labels = [""] + labels
```

Comment:
here instead of doing this hack you can override this in the [derived class](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/mpi/python_scripts/distributed_restart_utility.py)

=> check what [solution I came up with for similar things](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/mpi/python_scripts/distributed_restart_utility.py#L27-L31) and use sth in that direction